var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -101,6 +101,13 @@ class Petsc(Package, CudaPackage, ROCmPackage):
     variant("double", default=True, description="Switches between single and double precision")
     variant("complex", default=False, description="Build with complex numbers")
     variant("debug", default=False, description="Compile in debug mode")
+    # Logging redefines all MPI functions via macros. This is disabled
+    # because it can break standard-conforming C++ code since macros
+    # cannot handle commas in template arguments. See e.g.
+    # <https://github.com/AMReX-Codes/amrex/issues/3009> for a problem
+    # caused by this. I reported this problem to the PETSc maintainers,
+    # but there is not public list of bug reports available.
+    variant("log", default=False, description="Enable logging")
```

Comment:
[mentioned via e-mail] I think its best to deal with the amrex issue differently.

And wrt this PR - I think its fine to add in the "log" variant - using the current configure default - i.e "True"

Reply:
@balay Thanks for commenting!

See <https://github.com/AMReX-Codes/amrex/issues/3009> for a more complicated failure mod: Our code (part of the Einstein Toolkit) used both AMReX and PETSc in the same source file, and PETSc's changes to the MPI declarations made the AMReX code not compile. The AMReX CI infrastructure is not able to pick up this problem since they don't `#include <petsc.h>` into their header files, and they thus released a version of AMReX that is now not usable for us.

I don't think it would work to ask library authors whose library is used together with PETSc to add parentheses around MPI calls arguments that use templates. For example, to update the AMReX 22.11 in this way, we would now need to add a patch for AMReX 22.11 Spack recipe with this change.

I would prefer a way to avoid this that is automatic.

Reply:
Could you try modifying your code that include petsc.h to:

```
#define PETSC_HAVE_BROKEN_RECURSIVE_MACRO
#include <petsc.h>
#include <amrex.. - assuming it also does include petsc.h>
```

Reply:
I can do that.

(The AMReX include files do not include any PETSc include files.)

Reply:
@eschnett Any progress?

Reply:
I am currently using a version of PETSc with logging disabled. I assume, as @balay suggested, that modifying our PETSc-using code this way would avoid the problem as well. We can close this pull request.